### PR TITLE
Core/Spells: Fixed various paladin spells and removed various unused as well

### PR DESCRIPTION
**Pull request by matanshukry**
**Changes proposed:**

-  Fixed various paladin spells and removed various unused as well

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Issues addressed:**

**Tests performed:**

(Does it build, tested in-game, etc.)

- [x] It builds
- [x] Tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

### CODE REVIEW

File: src/server/scripts/Spells/spell_paladin.cpp [hunk 1]
```diff
@@ -746,219 +558,180 @@ class spell_pal_hand_of_sacrifice : public SpellScriptLoader
...
-                    {
-                        if (sSpellMgr->GetSpellInfo((*i)->GetAmount(), GetCastDifficulty()))
-                        {
-                            spellId = (*i)->GetAmount();
-                            break;
-                        }
-                    }
-                }
+// 40470 - Paladin Tier 6 Trinket
+class spell_pal_item_t6_trinket : public AuraScript
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_paladin.cpp:689]: legacy set effects were removed in wod or legion afaik

File: src/server/scripts/Spells/spell_paladin.cpp [hunk 2]
```diff
@@ -1430,34 +1032,55 @@ class spell_pal_t8_2p_bonus : public SpellScriptLoader
...
+    PrepareAuraScript(spell_pal_zeal);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_PALADIN_JUDGEMENT_PROT_RET_R3, SPELL_PALADIN_JUDGEMENT_GAIN_HOLY_POWER });
+    }
+
+    void HandleEffectProc(AuraEffect* aurEff, ProcEventInfo& /*procInfo*/)
+    {
+        Unit* caster = GetCaster();
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_paladin.cpp:1047]: caster may be null

File: src/server/scripts/Spells/spell_paladin.cpp (hunk 1 above)
Comment (Shauren) [reply src/server/scripts/Spells/spell_paladin.cpp:689]: This is not a set bonus effect, its a trinket (spell is literally named "Paladin Tier 6 Trinket")

**Comment by illfated** [comment]
I am guessing that your PR branch is behind the current TC master branch, hence the file conflict.

**Comment by matanshukry** [comment]
> I am guessing that your PR branch is behind the current TC master branch, hence the file conflict.

Sort of; just made specific new PRs split from this one to make this one look less horrible, which is what caused the conflicts. Will resolve in a minute.

File: src/server/scripts/Spells/spell_paladin.cpp (hunk 2 above)
Comment (Ovahlord) [reply src/server/scripts/Spells/spell_paladin.cpp]: GetTarget() is what you wanna use for auras that equal getcaster in 9/10 cases

Comment (matanshukry) [reply src/server/scripts/Spells/spell_paladin.cpp]: Yeah I actually fixed that already after @mdX7 comment. Realized I should of used GetTarget + corrected the validate method.

**Comment by Shauren** [comment]
I made some minor adjustments to some of them, one change remains to be done

Righteous Verdict needs to be removed from Templar's Verdict script and converted to a proc aura script (267610 has proc flags in SpellInfo)

**Comment by matanshukry** [comment]
@Shauren you used the Spell version of HasPowerTypeCost / CalcPowerCost on the spells, but is that what we want?

Specifically for selfless healer, when divine storm is free (due to another talent) - shouldn't it still proc?
Tooltip makes me think it should ("power spending abilities"), although currently it won't since the power cost of it will be 0.

**Comment by Shauren** [comment]
If you were worried about that then it's not a problem - all costs are in that container, even those that calculated to 0

**Comment by matanshukry** [comment]
> If you were worried about that then it's not a problem - all costs are in that container, even those that calculated to 0

I'll do the remaining change later on tonight then.
And thanks!

**Pull request merged**